src/newTSFeature/codeGenerator.ts

```diff
@@ -0,0 +1,100 @@
+import { existsSync, writeFile } from "fs";
+import path = require("path");
+import { getTSDataLayerRepositoryCodeTemplate } from "../code_generator_templates/ts/ts_data_layer_template";
+import { getTSDomainLayerCodeTemplate } from "../code_generator_templates/ts/ts_domain_layer_template";
+import { getTSDeserializerCodeTemplate, getTSOrchestratorCodeTemplate, getTSSerializerCodeTemplate } from "../code_generator_templates/ts/ts_gateway_layer_template";
+import { TSFeatureType } from "../utils/enums";
+
+export const generateTSCode = (
+
+    featureClassName: string,
+	snakeCaseFeatureName: string,
+	targetDirectory: string,
+    type: TSFeatureType,
+
+
+): void => {
+
+
+	if (type == TSFeatureType.NEW_OPERATION) {
+
+        const targetPath = `${targetDirectory}/${snakeCaseFeatureName}`
+
+
+		generateDataLayerCode(`${targetPath}/data/repository`, featureClassName, snakeCaseFeatureName);
+
+		generateDomainLayerCode(`${targetPath}/domain/repository/`, featureClassName,snakeCaseFeatureName);
```

Comment:
gateway code is not being generated here

Reply:
Yup I thought it wont be needed 

all the APIs related to student operations are written in the main studentOperationsFolder

So there would be gateway code for each API (named them as EXISTING_OPERATIONS ) for now

Reply:
cool